pkg/core/chain/chain.go

```diff
@@ -349,6 +372,23 @@ func (c *Chain) acceptSuccessiveBlock(blk block.Block, kadcastHeight byte) error
 		c.highestSeen = blk.Header.Height
 	}
 
+	// Guarantee that the execution time of the entire process of accepting the
+	// next valid block (which includes rusk calls, block verification and
+	// persistence) will never be less than N Milliseconds.
+	// This won't be applied in cases when:
+	// node is in out-of-sync mode.
+	// block time is higher that ConsensusTimeThreshold
+	if prevBlockTimestamp+config.ConsensusTimeThreshold > blk.Header.Timestamp {
+		maxDelayMilli := config.Get().Consensus.ThrottleMilli
+		if maxDelayMilli == 0 {
+			maxDelayMilli = 2000
+		}
+
+		if d, err := throttle(startTime, maxDelayMilli); err == nil {
+			log.WithField("height", blk.Header.Height).WithField("sleep_for", d.String()).Trace("throttled")
```

Comment:
Should `d` be used to wait? Cannot figure out where this duration is used

Reply:
Good catch. Forgot to commit it. It's in throttle function.